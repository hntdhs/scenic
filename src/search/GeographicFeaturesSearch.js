import React, { useEffect, useState } from "react";
import Checkbox from "./Checkbox";

function GeographicFeaturesSearch({ onChange }) {
    const categories = [ 'canyon', 'coastline', 'desert', 'forest', 'grasslands', 'lake', 'mountains', 'swamp', 'river', 'urban'];
    const [selected, setSelected] = useState([]);
    // didn't work properly without doing the useState and useEffect and changeHandler here and instead doing it in the SearchForm component where it's called
        
    useEffect(() => {
        onChange(selected)
    }, [selected]);

    const changeHandler = (props) => {
        if (props.checked) {
            const index = selected.indexOf(props.value)
            if (index > -1) {
                selected.splice(index, 1)
                setSelected(selected)
            } else {
                selected.push(props.value);
                setSelected(selected)
            }
            // this is creating an array, checking to be sure that what's selected isn't already in the array then adding
            // this is used in SearchForm so that's where the value prop is given - value={geoFeaturesSelect}
            
        }
    }
    return (
        <div>
            {categories.map((c,i) => (
                <div key={i} className="form-check form-check-inline">
                    <Checkbox onChange={changeHandler} name={c}/>
                    {/* <label key={c} htmlFor={c} class="form-check-label">{c}</label> */}
                    <label key={c} htmlFor="flexCheckChecked" class="form-check-label">{c}</label>
                </div>
            ))
    
            }
        </div>
    )
}
 

export default GeographicFeaturesSearch;