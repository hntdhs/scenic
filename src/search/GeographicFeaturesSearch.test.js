import React from "react";
import { render } from "@testing-library/react";
import GeographicFeaturesSearch from "./GeographicFeaturesSearch.js";
import { expect } from "@jest/globals";

it("matches snapshot", function () {
  const testHandlerMock = jest.fn(x => 42 + x);
    const { asFragment } = render(<GeographicFeaturesSearch onChange={testHandlerMock}/>);
    expect(asFragment()).toMatchSnapshot();
    expect(testHandlerMock).toBeCalled()
  });