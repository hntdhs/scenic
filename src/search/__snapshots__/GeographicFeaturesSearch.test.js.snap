// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="canyon"
          type="checkbox"
          value="canyon"
        />
        <label>
          canyon
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="coastline"
          type="checkbox"
          value="coastline"
        />
        <label>
          coastline
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="desert"
          type="checkbox"
          value="desert"
        />
        <label>
          desert
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="forest"
          type="checkbox"
          value="forest"
        />
        <label>
          forest
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="grasslands"
          type="checkbox"
          value="grasslands"
        />
        <label>
          grasslands
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="lake"
          type="checkbox"
          value="lake"
        />
        <label>
          lake
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="mountains"
          type="checkbox"
          value="mountains"
        />
        <label>
          mountains
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="swamp"
          type="checkbox"
          value="swamp"
        />
        <label>
          swamp
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="river"
          type="checkbox"
          value="river"
        />
        <label>
          river
        </label>
      </div>
    </div>
    <div
      class="form-check form-check-inline"
    >
      <div>
        <input
          class="form-check-input"
          id="flexCheckDefault"
          name="urban"
          type="checkbox"
          value="urban"
        />
        <label>
          urban
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
